### DISCUSSION TITLE
Stargate on existing cassandra data

### DISCUSSION THREAD

**Discussion by ybudweiser** [stargate/stargate#2610] [General] [answered]
to use the startgate REST and Document API must the data be ingested first via startgate's APis or can we query and search using the Document API on existing cassandra data that was perhaps sinked in via kafka connect to cassandra?

**Comment by ivansenic** [comment]
You need to ingest data first via the API. In addition, the Documents API is getting deprecated as we are building https://github.com/stargate/jsonapi.

**Reply by ybudweiser** [reply]
if we had data in kafka - that we want ingested into cassandra but queryable via stargate, you are saying we would not be able to a regular kafka connect to cassandra plugin. we would need to ingest it via the stargate json api?  is thier a java sdk client for jsonapi?

**Reply by ivansenic** [reply]
Yes exactly. There is no official Java SDK for the `jsonapi`, however it exposes a REST API as interface. So you would have to have a Kafka consumer that is inserting documents via the REST calls. The `insertMany` command is also available, if you want to have a batched approach.

Please note that `jsonapi` is still in development, with the preview release planned for August and GA in the autumn.

**Reply by ybudweiser** [reply]
so to be clear on nameing- when you say the jsonapi - is that this https://stargate.io/docs/latest/develop/dev-with-rest.html ?
or is the "Rest API" different then the jsonapi?

**Reply by ivansenic** [reply, marked as answer]
OK then there was a confusion.. Let me try to be more precise..

## REST API (https://stargate.io/docs/latest/develop/dev-with-rest.html)
This is not `jsonapi`, this is REST API for accessing the CQL data. For this API, you don't need to ingested data through it.. It works out of the box with the existing CQL tables you might have in Cassandra. Thus, if you want to use this you can use existing Kafka connector, insert data using CQL and then read/write after that with REST API. 

This is what we call *CQL-first* API. It means you can write/read using CQL and/or REST API at the same time. For example one app can write rows using CQL, and other app can read data using REST calls. Or vice versa. You work against table rows.

## Documents API (https://stargate.io/docs/latest/develop/dev-with-doc.html)
This is a special API, used for storing schema-less JSON documents into a Cassandra table. All documents must be inserted and read using using the API directly, as a document is parsed and stored to a Cassadnra table in a specific way (multiple rows per document).

## JSONAPI (https://github.com/stargate/jsonapi) `in-development`
Deprecates the Documents API, it's also used for storing schema-less JSON documents in Cassandra table, but the way how a document is stored differs (it's a row per document). Also set of features is different from Documents API. Also needs ingesting and reading documents through the API only.

@ybudweiser Would this clarify everything?

**Reply by ybudweiser** [reply]
Yes Perfect!
Thanks

**Comment by MatveiBOGH** [comment]


**Comment by BritvaBo** [comment]
thanks. got it

**Comment by jack99problems** [comment]
thank you a lot
